include/device.h

```diff
@@ -150,11 +150,18 @@ extern "C" {
 		.device_pm_control = (pm_control_fn),			  \
 		.config_info = (cfg_info)				  \
 	};								  \
+	static struct device_pm _CONCAT(__pm_, dev_name) __used           \
+							= {               \
+		.name = drv_name,					  \
```

Comment:
Could we either replace name with pointer to device or embed `device_pm` in `device`?
IMHO this would reduce work needed to traverse from one structure to another.

Reply:
the issue is, the device data is being put in init section which is read-only.

 static struct device _CONCAT(__device_, dev_name) __used          \
        **__attribute__((__section__(".init_" #level STRINGIFY(prio)))) = { \**

Due to this i had to use pointer here.

Reply:
Ok. Are you going to replace the name by the pointer to device structure?

Reply:
yes will fix it.